cmd/rehearse/main.go

```diff
@@ -0,0 +1,222 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+
+	// TODO: Solve this properly
+	// "github.com/davecgh/go-spew/spew"
+	"github.com/getlantern/deepcopy"
+	// "github.com/ghodss/yaml"
+	"github.com/sirupsen/logrus"
+
+	pjapi "k8s.io/test-infra/prow/apis/prowjobs/v1"
+	pjclientset "k8s.io/test-infra/prow/client/clientset/versioned"
+	"k8s.io/test-infra/prow/config"
+	prowflagutil "k8s.io/test-infra/prow/flagutil"
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/pjutil"
+
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+func getJobsToExecute(ca *config.Agent) config.JobConfig {
+	var youHaveOneJob config.Presubmit
+	for _, job := range ca.Config().Presubmits["openshift/ci-operator"] {
+		if job.Name == "pull-ci-openshift-ci-operator-master-build" {
+			youHaveOneJob = job
+			break
+		}
+	}
+
+	jobs := config.JobConfig{
+		Presubmits: map[string][]config.Presubmit{"openshift/ci-operator": []config.Presubmit{youHaveOneJob}},
+	}
+	return jobs
+}
+
+func makeRehearsalPresubmit(source *config.Presubmit, repo string, prNumber int) (*config.Presubmit, error) {
+	var rehearsal config.Presubmit
+	deepcopy.Copy(&rehearsal, source)
+
+	rehearsal.Name = fmt.Sprintf("rehearse-%d-%s", prNumber, source.Name)
+	rehearsal.Context = fmt.Sprintf("ci/rehearse/%s/%s", repo, strings.TrimPrefix(source.Context, "ci/prow/"))
+
+	if len(source.Spec.Containers) != 1 {
+		return nil, fmt.Errorf("Cannot rehearse jobs with more than 1 container in Spec")
+	}
+	container := source.Spec.Containers[0]
+
+	if len(container.Command) != 1 || container.Command[0] != "ci-operator" {
+		return nil, fmt.Errorf("Cannot rehearse jobs that have Command different from simple 'ci-operator'")
```

Comment:
Why?

Reply:
In case the job is not running a `ci-operator` we can manipulate the `$JOB_SPEC` and pass it to the generated prowjob.

Reply:
See the thread https://github.com/openshift/ci-operator-prowgen/pull/49#discussion_r243037220

We can certainly explore this, but at the moment directing the generated Prowjobs's `Refs` to point to a different repo would break the already working context reporting to the `openshift/release` PR.